File: Projects/codechef/XORAGN.java
package codechef;

import java.util.Arrays;
import java.util.Scanner;

class XORAGN {
    public static void main(String[] args) throws java.lang.Exception {
        Scanner stdin = new Scanner(System.in);
        int t = Integer.parseInt(stdin.nextLine());
        while (t-- > 0) {
            int n = Integer.parseInt(stdin.nextLine());
            System.out.println(
                    Arrays.stream(stdin.nextLine().split(" "))
                            .mapToInt(Integer::parseInt)
                            .map(x -> x * 2)
                            .reduce((x1, x2) -> x1 ^ x2)
                            .getAsInt()
            );
        }
    }
}

File: Projects/codechef/XORAGN.kt
package codechef

import java.io.IOException
import java.util.*

internal object XORAGN {
    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val stdin = Scanner(System.`in`)
        var t = stdin.nextLine().toInt()
        while (t-- > 0) {
            stdin.nextLine().toInt()
            println(
                stdin.nextLine()
                .split(" ")
                .map(String::toInt)
                .map { x: Int -> x * 2 }
                .reduce { x1: Int, x2: Int -> x1 xor x2 }
            )
        }
        stdin.close()
    }
}
